statsmodels/tsa/statespace/kalman_filter.py

```diff
@@ -1253,36 +1258,35 @@ def update_filter(self, kalman_filter):
                 # produce forecasts, but forecast errors and the forecast
                 # error covariance matrix will be zeros - make them nan to
                 # improve clarity of results.
-                if self.nmissing[t] == self.k_endog:
+                if self.nmissing[t] > 0:
+                    mask = ~self.missing[:, t].astype(bool)
```

Comment:
It's an integer. Do you know if I can safely use it without casting? The new numpy indexing changes have made me nervous.